src/Psalm/Internal/Analyzer/FunctionLikeAnalyzer.php

```diff
@@ -1995,6 +1995,7 @@ private function getFunctionInformation(
                 if (!$has_override_attribute
                     && $codebase->config->ensure_override_attribute
                     && $overridden_method_ids
+                    && $storage->visibility !== ClassLikeAnalyzer::VISIBILITY_PRIVATE
```

Comment:
Unfortunately this doesn't fully fix the issue.

Even if the trait method is `protected` or `public`, there's a false positive: https://psalm.dev/r/4dce1a9157 And in that code, it would still be incorrect to add an `Override` attribute, since the method in `A` doesn't override anything. The attribute causes an error at runtime.

I think ideally we'd not consider methods that come from traits, but I can't see an easy way to check for this.

Reply:
@edsrzf Thank you for reviewing the Pull request,

You're right!

> Methods from a used trait behave as if the method definition was copied and pasted into the target class. Specifically the `#[\Override]` attribute on a trait method requires the existence of a matching method in a parent class or implemented interface.

I'll take a closer look.